src/NuGet.Core/NuGet.Protocol.Core.v3/Resources/PackageUpdateResource.cs

```diff
@@ -275,18 +275,34 @@ private bool IsFileSource()
             return request;
         }
 
-        private void PushPackageToFileSystem(Uri sourceUri, string pathToPackage)
+        private async Task PushPackageToFileSystem(Uri sourceUri, 
+            string pathToPackage,
+            ILogger log, 
+            CancellationToken token)
         {
             string root = sourceUri.LocalPath;
             var reader = new PackageArchiveReader(pathToPackage);
             var packageIdentity = reader.GetIdentity();
 
-            //TODD: support V3 repo style if detect it is
-            var pathResolver = new PackagePathResolver(root, useSideBySidePaths: true);
-            var packageFileName = pathResolver.GetPackageFileName(packageIdentity);
-
-            var fullPath = Path.Combine(root, packageFileName);
-            File.Copy(pathToPackage, fullPath, overwrite: true);
+            if (IsV2LocalRepository(root))
```

Comment:
This is true if the folder does not exist, but I don't see it create the folder afterwards. I thought new folders would use v3?